fog/ledger/server/tests/router_integration.rs

```diff
@@ -115,22 +134,23 @@ fn seed_block_provider(block_provider: &mut LocalBlockProvider<LedgerDB>) {
     add_block_to_ledger(block_provider, &recipients, &[], &mut rng);
 }
 
-fn populate_block_provider(
+fn populate_block_provider<'a>(
     block_provider: &mut LocalBlockProvider<LedgerDB>,
-    blocks_config: &BlockConfig,
+    blocks_config: impl IntoIterator<Item = &'a HashMap<PublicAddress, Vec<KeyImage>>>,
 ) {
     let mut rng = thread_rng();
 
-    for block in blocks_config {
+    for block in blocks_config.into_iter() {
         let recipients: Vec<_> = block.keys().cloned().collect();
         let key_images: Vec<_> = block.values().flat_map(|x| x.clone()).collect();
 
         add_block_to_ledger(block_provider, &recipients, &key_images, &mut rng);
     }
 
-    // The stores are running on separate threads. We wait twice as long as
-    // their POLL_INTERVAL to ensure they've had time to process the new blocks
-    std::thread::sleep(POLL_INTERVAL * 2);
+    // The stores are running on separate threads. We wait 3(stores) + 1
+    // their POLL_INTERVAL to ensure they've all had time to process the new 
+    // blocks
+    std::thread::sleep(POLL_INTERVAL * 4);
```

Comment:
not fond of bumping more, but I was running into issues where not all of the stores were being updated in time.